src/tests/tests.cpp

```diff
@@ -32,9 +32,14 @@ namespace Botan_Tests {
 
 void Test::Result::merge(const Result& other, bool ignore_test_name)
    {
-   if(!ignore_test_name && who() != other.who())
+   if(!ignore_test_name)
       {
-      throw Test_Error("Merging tests from different sources");
+      if(who() != other.who())
+         throw Test_Error("Merging tests from different sources");
+
+      // When deliberately merging results with different names, the code location is
+      // likely inconsistent and must be discarded.
```

Comment:
If they are inconsistent would it maybe make sense to also unset the maybe already set `m_where`? Currently only the location of `other` is discarded.

Reply:
Good point. I changed it around so that `m_where` is now discarded if names don't match and `other.m_where` is taken if names match.